### PR TITLE
feat: add biome linter & formatter

### PR DESCRIPTION
**Pull request by o-az**
This pr
- replaces eslint and prettier in all workspaces with biomejs,
- adds biome ci checks,
- runs biome and fixes some lint issues,
- adds pre-commit hook to run lint & format & typecheck.

still using `prettier` in one place: https://github.com/atomiks/rehype-pretty-code/blob/c23ec2006588d96fcdbb83a9cefc10accd86aafb/test/fixtures.test.js#L12

will replace later.

### CODE REVIEW

**Comment by o-az** [comment]
> Can we keep the previous formatting style options? (semi, arrowParens for arrow functions)

done

File: biome.json [hunk 1]
```diff
@@ -87,12 +87,12 @@
       "lineWidth": 100,
       "indentWidth": 2,
       "indentStyle": "space",
-      "quoteStyle": "single",
-      "trailingComma": "none",
-      "semicolons": "asNeeded",
+      "quoteStyle": "double",
```
Comment (atomiks) [inline biome.json:90]:
```suggestion
      "quoteStyle": "single",
```

Comment (o-az) [reply biome.json]: done

File: website/src/app/page.tsx [hunk 2]
```diff
@@ -76,7 +69,7 @@ export default function Home() {
         </main>
       </div>
       <footer
-        className="w-full text-amber-100/70 !max-w-none prose dark:prose-invert text-center pb-[10rem] pt-[35rem] -mt-[25rem]"
+        className='w-full text-amber-100/70 !max-w-none prose dark:prose-invert text-center pb-[10rem] pt-[35rem] -mt-[25rem]'
```
Comment (atomiks) [inline website/src/app/page.tsx:72]: Any way to keep double quotes for JSX?

Comment (o-az) [reply website/src/app/page.tsx]: done

**Pull request merged**